pkg/csi/networkBuilder.go

```diff
@@ -268,10 +268,7 @@ func createNetFromLayoutConfig(conf NetworkLayoutConfiguration) (*IPV4Network, e
 			}
 			subnet.FullName = fmt.Sprintf("%v Bootstrap DHCP Subnet", tempNet.Name)
 			if tempNet.Name == "NMN" || tempNet.Name == "HMN" || tempNet.Name == "CMN" || tempNet.Name == "CAN" || tempNet.Name == "CHN" {
-				if tempNet.Name == "CMN" {
-					subnet.CIDR = *cmnCIDR
-					subnet.Gateway = net.ParseIP(v.GetString("cmn-gateway"))
```

Comment:
Does the gateway not get set to cmn-gateway input anymore?   Or does that happen elsewhere?

Reply:
Okay, I'd like to see the DHCP ranges that get generated.   I want to see if that introduces overlap with the static/dynamic pools.  Do you have an example of the generated files somewhere again?

Reply:
the cmn-gateway overrode the conf. logic and created the bug here.

Reply:
It looks like the code will now always use the first IP for the gateway.   If we want/need to do that then we should remove the requirement for the cmn-gateway input.   That is currently a required input but will be ignored.
I think we'll also need to make a change to dnsmasq.go since that does currently use cmn-gateway.

Reply:
@johren-hpe I have removed cmn-gateway as a cli param as requested.  Seems like an output no-op as expected.

Reply:
Something is still off with the gateways generated in data.json.   When @spillerc-hpe used this branch on hela this morning, the generated ipam gateways iin data.json for MTL, HMN, and CMN are showing up as the IP address of m001 for that network.

```
        "cmn": {
          "gateway": "10.103.0.28",
          "ip": "10.103.0.28/25",
          "parent_device": "bond0",
          "vlanid": 7
        },
        "hmn": {
          "gateway": "10.254.1.22",
          "ip": "10.254.1.22/17",
          "parent_device": "bond0",
          "vlanid": 4
        },
        "mtl": {
          "gateway": "10.1.1.11",
          "ip": "10.1.1.11/16",
          "parent_device": "bond0",
          "vlanid": 0
        },
```

Reply:
This could be a latent bug that was always there for HMN and MTL but wasn't really an issue if those gateways are not used anywhere on the NCNs.  But it be comes a problem with CMN since that is used as the default route.

The weird thing is that the write_files section of that file does have the correct HMN gateway.
```
            "write_files": [
                {
                    "content": "10.252.0.0/22 10.252.0.1 - bond0.nmn0\n10.1.0.0/16 10.252.0.1 - bond0.nmn0\n10.92.100.0/24 10.252.0.1 - bond0.nmn0\n",
                    "owner": "root:root",
                    "path": "/etc/sysconfig/network/ifroute-bond0.nmn0",
                    "permissions": "0644"
                },
                {
                    "content": "10.254.0.0/22 10.254.0.1 - bond0.hmn0\n10.94.100.0/24 10.254.0.1 - bond0.hmn0\n",
                    "owner": "root:root",
                    "path": "/etc/sysconfig/network/ifroute-bond0.hmn0",
                    "permissions": "0644"
                }
            ]
```

Reply:
I've moved the gateway changes to https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-3657 and reverted it here.  Pretty sure you're right with "latent bug".

Reply:
We'll just have to make sure that the cmn-gateway that is provided for now is always the first IP in the subnet.   Otherwise SLS will not match with what is in dnsmasq and basecamp.